base/server/src/org/dogtagpki/server/rest/SecurityDomainHostService.java

```diff
@@ -0,0 +1,177 @@
+        }
+
+        String type = matcher.group(1);
+        logger.debug("SecurityDomainService: type: " + type);
+
+        String hostname = matcher.group(2);
+        logger.debug("SecurityDomainService: hostname: " + hostname);
+
+        String port = matcher.group(3);
+        logger.debug("SecurityDomainService: port: " + port);
+
+        String unsecurePort = host.getPort();
+        logger.debug("SecurityDomainService: unsecure port: " + unsecurePort);
+
+        String eeCAPort = host.getSecureEEClientAuthPort();
+        logger.debug("SecurityDomainService: EE port: " + eeCAPort);
+
+        String adminSecurePort = host.getSecureAdminPort();
+        logger.debug("SecurityDomainService: admin port: " + adminSecurePort);
+
+        String agentSecurePort = host.getSecureAgentPort();
+        logger.debug("SecurityDomainService: agent port: " + agentSecurePort);
+
+        String domainManager = host.getDomainManager();
+        logger.debug("SecurityDomainService: domain manager: " + domainManager);
+
+        String clone = host.getClone();
+        logger.debug("SecurityDomainService: clone: " + clone);
+
+        SecurityDomainProcessor processor = new SecurityDomainProcessor(getLocale(headers));
+        String status = processor.addHost(
+                hostID,
+                type,
+                hostname,
+                port,
+                unsecurePort,
+                eeCAPort,
+                adminSecurePort,
+                agentSecurePort,
+                domainManager,
+                clone);
+        logger.debug("SecurityDomainService: status: " + status);
+
+        if (!SecurityDomainProcessor.SUCCESS.equals(status)) {
+            throw new PKIException("Unable to remove security domain host: " + hostID);
+        }
+
+        return createNoContentResponse();
+    }
+
+    @Override
+    public Response removeHost(String hostID) throws Exception {
+
+        logger.info("SecurityDomainService: Removing security domain host \"" + hostID + "\"");
+
+        // Host ID: <type> <hostname> <port>
+        Pattern pattern = Pattern.compile("^(\\S+) (\\S+) (\\d+)$");
+        Matcher matcher = pattern.matcher(hostID);
+
+        if (!matcher.find()) {
+            throw new BadRequestException("Invalid security domain host: " + hostID);
+        }
+
+        String type = matcher.group(1);
+        logger.debug("SecurityDomainService: type: " + type);
+
+        String hostname = matcher.group(2);
+        logger.debug("SecurityDomainService: hostname: " + hostname);
+
+        String port = matcher.group(3);
+        logger.debug("SecurityDomainService: port: " + port);
+
+        SecurityDomainProcessor processor = new SecurityDomainProcessor(getLocale(headers));
+        String status = processor.removeHost(type, hostname, port);
+        logger.debug("SecurityDomainService: status: " + status);
+
+        if (!SecurityDomainProcessor.SUCCESS.equals(status)) {
```

Comment:
Executing `securitydomain-host-del <host ID>` twice does not throw any warning/errors

Reply:
Yes, same thing with add, the current code is ignoring deletion errors:
https://github.com/dogtagpki/pki/blob/master/base/server/src/com/netscape/cms/servlet/csadmin/SecurityDomainProcessor.java#L638

Let's fix this in a separate PR.

Reply:
Agreed!